dlt/destinations/impl/sqlalchemy/merge_job.py

```diff
@@ -120,6 +120,7 @@ def gen_merge_sql(
         )
 
         dedup_sort = get_dedup_sort_tuple(root_table)  # column_name, 'asc' | 'desc'
+        skip_dedup = root_table.get("x-stage-data-deduplicated", False)
```

Comment:
at some point we should collect these hints in a const file and not use magic strings everywhere :)